vault/resource_transit_secret_backend_key.go

```diff
@@ -317,24 +332,45 @@ func transitSecretBackendKeyRead(d *schema.ResourceData, meta interface{}) error
 		return fmt.Errorf("expected min_encryption_version %q to be a number, and it isn't", secret.Data["min_encryption_version"])
 	}
 
-	ikeys := secret.Data["keys"].(map[string]interface{})
+	ikeys := secret.Data["keys"]
 	keys := []interface{}{}
-	for _, v := range ikeys {
-		// Data structure of "keys" differs depending on encryption key type. Sometimes it's a single integer hash,
-		// and other times it's a full map of values describing the key version's creation date, name, and public key.
-
-		if sv, ok := v.(map[string]interface{}); ok { // for key types of rsa-2048, rsa-3072, rsa-4096, ed25519, ecdsa-p256, ecdsa-p384 or ecdsa-p521
-			keys = append(keys, sv)
-		} else if sv, ok := v.(json.Number); ok { // for key types of aes128-gcm96, aes256-gcm96 or chacha20-poly1305
-			m := make(map[string]interface{})
-			m["id"] = sv
-			keys = append(keys, m)
+	if ikeys != nil || secret.Data["type"] != "hmac" { // hmac type does not return keys
+		ikeys := secret.Data["keys"].(map[string]interface{})
+		for _, v := range ikeys {
+			// Data structure of "keys" differs depending on encryption key type. Sometimes it's a single integer hash,
+			// and other times it's a full map of values describing the key version's creation date, name, and public key.
+			if sv, ok := v.(map[string]interface{}); ok { // for key types of rsa-2048, rsa-3072, rsa-4096, ed25519, ecdsa-p256, ecdsa-p384 or ecdsa-p521
+				keys = append(keys, sv)
+			} else if sv, ok := v.(json.Number); ok { // for key types of aes128-gcm96, aes256-gcm96 or chacha20-poly1305
+				m := make(map[string]interface{})
+				m["id"] = sv
+				keys = append(keys, m)
+			}
 		}
+
 	}
 
 	if err := d.Set("keys", keys); err != nil {
 		return err
 	}
+
+	// On read, key_size will be nil if the encryption key type is not HMAC. Choosing not to set it in those cases.
+	key_size := secret.Data["key_size"]
+	if key_size != nil || secret.Data["type"] == "hmac" {
```

Comment:
Same comment on checking the vault server version is 1.12 or greater which is when support was added for key_size and type hmac.

Reply:
I wrapped this whole block in a version check.